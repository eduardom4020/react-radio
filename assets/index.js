import RussiaBG from './russia.jpg';
import DribbbledBG from './dribbbled.jpg';
import DogeBG from './doge.jpg';
import BalladsBG from './ballads.jpg';
import MaximumBG from './maximum.jpg';

import Switch from './switch.png';
import Back from './back-arrow.png';

export const radioPutinLogo = RussiaBG;
export const radioDribbbledLogo = DribbbledBG;
export const radioDogeLogo = DogeBG;
export const radioBalladsLogo = BalladsBG;
export const radioMaximumLogo = MaximumBG;

export const iconSwitch = Switch;
export const iconBack = Back;