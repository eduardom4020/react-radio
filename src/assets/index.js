import RussiaBG from './russia.jpg';
import DribbbledBG from './dribbbled.jpg';
import DogeBG from './doge.jpg';
import BalladsBG from './ballads.jpg';
import MaximumBG from './maximum.jpg';

import Switch from './switch.png';
import Back from './back-arrow.png';
import Plus from './plus.png';
import Minus from './minus.png';

export const radioPutinLogo = RussiaBG;
export const radioDribbbledLogo = DribbbledBG;
export const radioDogeLogo = DogeBG;
export const radioBalladsLogo = BalladsBG;
export const radioMaximumLogo = MaximumBG;

export const iconSwitch = Switch;
export const iconBack = Back;
export const iconPlus = Plus;
export const iconMinus = Minus;