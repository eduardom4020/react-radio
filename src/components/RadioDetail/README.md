# RadioDetail
>This component is a View that renders a ImageButton between two IconButtons. All of these are clickable.

### Props
These are the props accepted by the component, pay attention to the description, inputs and behaviour:

---

##### imageTag *(string)*
*In the root of the project you should have an assets folder with images. This folder exports all the images that can be used by this component. The tag props is the name of the exported Module*.
**example**: 
```html
<RadioDetail imageTag='loadedImage1' />
```

---

##### onTouchImage *(function)*
*A callback function to when the ImageButton is clicked.*
**example**: 
```html
<RadioDetail onTouchImage={() => console.log('Clicked!')} />
```

---

##### style *(object)*
*CSS styles to be applyied in the View wapper wich contains the three images.*
**example**: 
```html
<RadioDetail style={{flexDirection: 'column'}} />, *it should change displacement of contents from horizontal to vertical.*
```

---