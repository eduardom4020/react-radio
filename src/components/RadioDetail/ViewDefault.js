import React from 'react';
import { View, Image, TouchableWithoutFeedback } from 'react-native';
import Minus from '~/assets/minus.png';
import Plus from '~/assets/plus.png';
import * as Images from '~/assets';
import IconButton from '~/src/components/IconButton';

const RadioDetailViewDefault = props => {
    const { imageTag='', style={}, onTouchImage } = props;

    return (
        <View
            style={{
                width: '100%',
                height: '100%',
                flexDirection: 'row',
                justifyContent: 'space-around',
                alignItems: 'center',
                overflow: 'hidden',
                ...style
            }}
        >
            <IconButton src={Minus} />
            <TouchableWithoutFeedback onPress={onTouchImage}>
                <Image 
                    source={Images[imageTag]} 
                    style={{
                        width: 192, 
                        height: '100%',
                        maxHeight: 192,
                        borderRadius: 192 / 2,
                        borderWidth: 2,
                        borderColor: '#a3acbe'
                    }}
                />
            </TouchableWithoutFeedback>
            <IconButton src={Plus} />
        </View>
    );
};

export default RadioDetailViewDefault;