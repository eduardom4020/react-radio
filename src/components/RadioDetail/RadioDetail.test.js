import React from 'react';
import { iconSwitch } from '~/assets';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import RadioDetail from '~/src/components/RadioDetail';

describe('RadioDetail', () => {
    test('Render?', () => {
        const component = shallow(
            <RadioDetail />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with valid tag?', () => {
        const componentObject = shallow(
            <RadioDetail imageTag='radioDogeLogo' />
        );

        expect(toJson(
            componentObject, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with invalid tag?', () => {
        const componentObject = shallow(
            <RadioDetail imageTag={{}} />
        );

        expect(toJson(
            componentObject, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
        
        const componentBool = shallow(
            <RadioDetail imageTag={true} />
        );

        expect(toJson(
            componentBool, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();

        const componentNumber = shallow(
            <RadioDetail imageTag={1} />
        );

        expect(toJson(
            componentNumber, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();

        const componentString = shallow(
            <RadioDetail imageTag='image' />
        );

        expect(toJson(
            componentString, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });
});