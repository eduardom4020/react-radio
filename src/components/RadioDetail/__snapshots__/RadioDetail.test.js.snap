// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioDetail Render with invalid tag? 1`] = `
<Styled(View)>
  <IconButtonViewDefault
    size={32}
    src={
      Object {
        "testUri": "../../../assets/minus.png",
      }
    }
    tintColor="#a3acbe"
  />
  <ImageButtonViewDefault
    tag={Object {}}
  />
  <IconButtonViewDefault
    size={32}
    src={
      Object {
        "testUri": "../../../assets/plus.png",
      }
    }
    tintColor="#a3acbe"
  />
</Styled(View)>
`;

exports[`RadioDetail Render with invalid tag? 2`] = `
<Styled(View)>
  <IconButtonViewDefault
    size={32}
    src={
      Object {
        "testUri": "../../../assets/minus.png",
      }
    }
    tintColor="#a3acbe"
  />
  <ImageButtonViewDefault
    tag={true}
  />
  <IconButtonViewDefault
    size={32}
    src={
      Object {
        "testUri": "../../../assets/plus.png",
      }
    }
    tintColor="#a3acbe"
  />
</Styled(View)>
`;

exports[`RadioDetail Render with invalid tag? 3`] = `
<Styled(View)>
  <IconButtonViewDefault
    size={32}
    src={
      Object {
        "testUri": "../../../assets/minus.png",
      }
    }
    tintColor="#a3acbe"
  />
  <ImageButtonViewDefault
    tag={1}
  />
  <IconButtonViewDefault
    size={32}
    src={
      Object {
        "testUri": "../../../assets/plus.png",
      }
    }
    tintColor="#a3acbe"
  />
</Styled(View)>
`;

exports[`RadioDetail Render with invalid tag? 4`] = `
<Styled(View)>
  <IconButtonViewDefault
    size={32}
    src={
      Object {
        "testUri": "../../../assets/minus.png",
      }
    }
    tintColor="#a3acbe"
  />
  <ImageButtonViewDefault
    tag="image"
  />
  <IconButtonViewDefault
    size={32}
    src={
      Object {
        "testUri": "../../../assets/plus.png",
      }
    }
    tintColor="#a3acbe"
  />
</Styled(View)>
`;

exports[`RadioDetail Render with valid tag? 1`] = `
<Styled(View)>
  <IconButtonViewDefault
    size={32}
    src={
      Object {
        "testUri": "../../../assets/minus.png",
      }
    }
    tintColor="#a3acbe"
  />
  <ImageButtonViewDefault
    tag="radioDogeLogo"
  />
  <IconButtonViewDefault
    size={32}
    src={
      Object {
        "testUri": "../../../assets/plus.png",
      }
    }
    tintColor="#a3acbe"
  />
</Styled(View)>
`;

exports[`RadioDetail Render? 1`] = `
<Styled(View)>
  <IconButtonViewDefault
    size={32}
    src={
      Object {
        "testUri": "../../../assets/minus.png",
      }
    }
    tintColor="#a3acbe"
  />
  <ImageButtonViewDefault
    tag=""
  />
  <IconButtonViewDefault
    size={32}
    src={
      Object {
        "testUri": "../../../assets/plus.png",
      }
    }
    tintColor="#a3acbe"
  />
</Styled(View)>
`;
